packages/react-reconciler/src/ReactFiberWorkLoop.new.js

```diff
@@ -3001,11 +3002,13 @@ if (__DEV__ && replayFailedUnitOfWorkWithInvokeGuardedCallback) {
       return originalBeginWork(current, unitOfWork, lanes);
     } catch (originalError) {
       if (
-        originalError !== null &&
-        typeof originalError === 'object' &&
-        typeof originalError.then === 'function'
+        hydrationDidSuspendOrErrorDEV() ||
```

Comment:
What’s the timing of this flag being set? We want to be able to still break on the first error so if this flag is already set it wouldn’t.

I think you might need to read it before beginWork to see if it was already set before.

Reply:
Begin work catches first and then handleError delegates to throwException where the flag is set. Timing too fragile?